dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderResource.java

```diff
@@ -70,41 +61,24 @@ public final Response createFolders(@Context final HttpServletRequest httpServle
                                         @Context final HttpServletResponse httpServletResponse,
                                         final List<String> paths,
                                         @PathParam("siteName") final String siteName) {
-        Response response = null;
-        final InitDataObject initData = this.webResource.init(null, httpServletRequest, httpServletResponse, true, null);
+
+        final InitDataObject initData =
+                new WebResource.InitBuilder(webResource)
+                        .requiredBackendUser(true)
+                        .requiredFrontendUser(false)
+                        .requestAndResponse(httpServletRequest, httpServletResponse)
+                        .init();
+
         final User user = initData.getUser();
-        final List<Map<String, Object>> folderResults;
 
         try {
-            Locale locale = LocaleUtil.getLocale(user, httpServletRequest);
-            FolderHelper.FolderResults results = folderHelper.createFolders(paths, siteName, user);
-            folderResults = results.folders
-                    .stream()
-                    .map(folder -> {
-                        try {
-                            return folder.getMap();
-                        } catch (Exception e) {
-                            Logger.error(this, "Data Exception while converting to map", e);
-                            throw new DotRuntimeException("Data Exception while converting to map", e);
-                        }
-                    })
-                    .collect(Collectors.toList());
-            ;
+            final List<Map<String, Object>> createdFolders = folderHelper.createFolders(paths, siteName, user);
 
-            response = Response.ok(new ResponseEntityView
-                    (map("result", folderResults),
-                            results.getErrorEntities(), null,
-                            this.i18NUtil.getMessagesMap(locale, "Invalid-option-selected",
-                                    "cancel")
-                    )).build(); // 200
-        } catch (Exception e) { // this is an unknown error, so we report as a 500.
-            Logger.error(this, "Error handling Save Folder Post Request", e);
-            if (ExceptionUtil.causedBy(e, DotSecurityException.class)) {
-                throw new ForbiddenException(e);
-            }
-            response = ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+            return Response.ok(new ResponseEntityView(createdFolders)).build(); // 200
+        } catch (Exception e) {
+            Logger.error(this.getClass(),"Exception saving/creating folders.", e);
+            return ResponseUtil.mapExceptionResponse(e);
```

Comment:
here we should use the Exception Mapper

Reply:
do you have an example of the exception mapper? Because I based on the AppsResource that is the latest resource created and this is how is handling the exceptions.

Reply:
done